src/main/java/net/datafaker/providers/base/Internet.java

```diff
@@ -66,7 +67,64 @@ public String domainSuffix() {
         return resolve("internet.domain_suffix");
     }
 
+    /**
+     * Returns a web domain.
+     * 
+     * @return a domain in the form "www.example.com"
+     * @deprecated Since 2.0.0 
+     * @see #webdomain()
+     * @see #weburl()
+     */
+    @Deprecated(since="2.0.0",forRemoval = true) 
     public String url() {
+        return webdomain();
+    }
+
+    /**
+     * Returns a string representing a web URL, randomly including: http/https scheme, creds, port, path
+     * elements (2 or none), file element (1 or none), params (2 or none), anchor (1 or none).
+     * 
+     * @return a web URL
+     * @since 2.0.0
+     */
+    public String weburl() {
+        return weburl(faker.random().nextBoolean(), faker.random().nextBoolean(),
+                faker.random().nextBoolean(), faker.random().nextBoolean(),
+                faker.random().nextBoolean(), faker.random().nextBoolean(),
+                faker.random().nextBoolean());
```

Comment:
what about
```suggestion
       final byte[] bts = faker.random().nextRandomBytes(7);
        return weburl(bts[0] % 2 == 0, bts[1] % 2 == 0,
                bts[2] % 2 == 0, bts[3] % 2 == 0,
                bts[4] % 2 == 0, bts[5] % 2 == 0,
                bts[6] % 2 == 0);
```
it will be faster since there is only one call to `random`

Reply:
Sure, I'm good with that. I hadn't put any thought into the performance.